fili-core/src/main/java/com/yahoo/bard/webservice/web/util/CacheService.java

```diff
@@ -0,0 +1,173 @@
+import static com.yahoo.bard.webservice.web.handlers.PartialDataRequestHandler.getPartialIntervalsWithDefault;
+import static com.yahoo.bard.webservice.web.handlers.VolatileDataRequestHandler.getVolatileIntervalsWithDefault;
+
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
+import com.yahoo.bard.webservice.application.MetricRegistryFactory;
+import com.yahoo.bard.webservice.data.cache.TupleDataCache;
+import com.yahoo.bard.webservice.druid.model.query.DruidAggregationQuery;
+import com.yahoo.bard.webservice.logging.RequestLog;
+import com.yahoo.bard.webservice.logging.blocks.BardQueryInfo;
+import com.yahoo.bard.webservice.metadata.QuerySigningService;
+import com.yahoo.bard.webservice.web.handlers.RequestContext;
+import com.yahoo.bard.webservice.web.responseprocessors.ResponseProcessor;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+
+/**
+ * A utility class to assist with caching.
+ */
+public class CacheService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CacheService.class);
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+    private final long maxDruidResponseLengthToCache = SYSTEM_CONFIG.getLongProperty(
+            SYSTEM_CONFIG.getPackageVariableName(
+                    "druid_max_response_length_to_cache"
+            ),
+            Long.MAX_VALUE
+    );
+    private static final MetricRegistry REGISTRY = MetricRegistryFactory.getRegistry();
+    public static final Meter CACHE_HITS = REGISTRY.meter("queries.meter.cache.hits");
+    public static final Meter CACHE_POTENTIAL_HITS = REGISTRY.meter("queries.meter.cache.potential_hits");
+    public static final Meter CACHE_MISSES = REGISTRY.meter("queries.meter.cache.misses");
+    public static final Meter CACHE_REQUESTS = REGISTRY.meter("queries.meter.cache.total");
+
+
+    /**
+     * Read cache.
+     *
+     * @param context The context data from the request processing chain
+     * @param dataCache The cache instance
+     * @param druidQuery The query being processed
+     * @param querySigningService The service to generate query signatures
+     * @param cacheKey Key into which to write a cache entry
+     * @return Response
+     */
+    public String readCache(
+            RequestContext context,
+            TupleDataCache<String, Long, String> dataCache,
+            QuerySigningService<Long> querySigningService,
+            DruidAggregationQuery<?> druidQuery,
+            String cacheKey
+    ) {
+        final TupleDataCache.DataEntry<String, Long, String> cacheEntry = dataCache.get(cacheKey);
```

Comment:
I think you should construct this with DataCache and QuerySigning Service and ObjectMapper.

Also, if we're passing the druidQuery in, we can move the cache key building into this class.

that would leave a simplified call of:
```suggestion
    /**
     * Read cache.
     *
     * @param context The context data from the request processing chain
     * @param druidQuery The query being processed

     * @return Response
     */
    public String readCache(
            RequestContext context,
            DruidAggregationQuery<?> druidQuery
    ) throws JsonProcessingException {
        final TupleDataCache.DataEntry<String, Long, String> cacheEntry = dataCache.get(getKey(druidQuery));
```